python/ccf/ledger.py

```diff
@@ -149,6 +164,200 @@ def _byte_read_safe(file, num_of_bytes):
     return ret
 
 
+class NodeStatus(IntEnum):
+    """These are the corresponding status meanings from the ccf.nodes table"""
+
+    pending = 0
+    trusted = 1
+    retired = 2
+
+
+class TxBundleInfo(NamedTuple):
+    """Bundle for transaction information required for validation """
+
+    merkle_tree: MerkleTree
+    existing_root: bytes
+    node_cert: bytes
+    signature: bytes
+    node_activity: dict
+    signing_node: int
```

Comment:
```suggestion
    signing_node: bytes
```

Node ids are going to be hex digests of their public key shortly, the fact that they are integers at the moment isn't going to last, they're best treated as opaque strings.